src/sources/util/grpc/mod.rs

```diff
@@ -51,10 +56,48 @@ where
         .layer(DecompressionAndMetricsLayer)
         .add_service(service)
         .serve_with_incoming_shutdown(stream, shutdown.map(|token| tx.send(token).unwrap()))
-        .in_current_span()
         .await?;
 
     drop(rx.await);
 
     Ok(())
 }
+
+pub fn build_grpc_trace_layer(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏same here